src/node/internal/internal_performance.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
+
+import {
+  ERR_METHOD_NOT_IMPLEMENTED,
+  ERR_ILLEGAL_CONSTRUCTOR,
+  ERR_MISSING_ARGS,
+} from 'node-internal:internal_errors';
+import { PerformanceEntry } from 'node-internal:internal_performance_entry';
+import { validateThisInternalField } from 'node-internal:validators';
+import { kEnumerableProperty } from 'node-internal:internal_utils';
+import { inspect } from 'node-internal:internal_inspect';
+import type { InspectOptions } from 'node-internal:internal_inspect';
+import type {
+  Performance as _Performance,
+  PerformanceMark as _PerformanceMark,
+  PerformanceNodeTiming,
+  MarkOptions,
+  EventLoopUtilization,
+} from 'node:perf_hooks';
+
+const kInspect = inspect.custom;
+const kPerformanceBrand = Symbol('performance');
+const kInitialize = Symbol('initialize');
+
+export class Performance extends EventTarget implements _Performance {
```

Comment:
Rather the introducing a new class, I wonder if it makes sense to augment the existing `globalThis.Performance` with the new APIs. We can have the internal C++ class extend from `EventTarget` and have it so that when `perf_hooks` is imported, we extend the `Performance.prototype` with the additional methods lazily.

Reply:
Specifically, let's make the following changes here:

* No new separate `Performance` class. Instead, extend the existing class defined in `global-scope.h` (likely best to move it out to a separate `api/performance.h` header and corresponding implementation file. Even if most of the new functions are non-ops, expose them there by default if they are web platform standard apis, and conditionally when `nodejs_compat`/`enable_nodejs_perf_hooks_module` flags are set.
* Move the implementations of `PerformanceEntry`, `PerformanceMark`, PerformanceMeasure`, `PerformanceObserver`, `PerformanceObserverEntryList`, `PerformanceResourceTiming` to C++, expose these as globals unconditionally, even if they are non-ops. Re-export them via `perf_hooks`.